stdlib/LinearAlgebra/test/symmetric.jl

```diff
@@ -462,6 +467,14 @@ end
     end
 end
 
+# bug identified in PR #52318: dot products of quaternionic Hermitian matrices,
+# or any number type where conj(a)*conj(b) ≠ conj(a*b):
+@testset "dot Hermitian quaternion #52318" begin
+    A, B = [Quaternion.(randn(3,3), randn(3, 3), randn(3, 3), randn(3,3)) |> t -> t + t' for i in 1:2]
+    @test A == Hermitian(A) && B == Hermitian(B)
+    @test dot(A,B) ≈ dot(Hermitian(A),Hermitian(B))
```

Comment:
I'd like to propose to add a corresponding symmetric test. It is currently passing, but I'm not sure if the explicit realification is still valid without also restricting `Symmetric` to commutative eltypes.

```suggestion
    A, B = [Quaternion.(randn(3,3), randn(3, 3), randn(3, 3), randn(3,3)) |> t -> t + t' for i in 1:2]
    @test A == Hermitian(A) && B == Hermitian(B)
    @test dot(A, B) ≈ dot(Hermitian(A), Hermitian(B))
    A, B = [Quaternion.(randn(3,3), randn(3, 3), randn(3, 3), randn(3,3)) |> t -> t + transpose(t) for i in 1:2]
    @test A == Symmetric(A) && B == Symmetric(B)
    @test dot(A, B) ≈ dot(Symmetric(A), Symmetric(B))
```

Reply:
Indeed, I noticed that later.